modules/optimization/include/utils/ParameterMesh.h

```diff
@@ -18,25 +18,45 @@
 #include "libmesh/explicit_system.h"
 #include "libmesh/point_locator_base.h"
 
+// fixme I can't get the forward declaration to work so I'm including the header.
+#include "libmesh/exodusII_io.h"
+
+// using namespace libMesh;
+// namespace libMesh
+// {
+// class ExodusII_IO;
+// }
+
 class ParameterMesh
 {
 public:
-  ParameterMesh(const FEType & param_type, const std::string & exodus_mesh);
-
-  dof_id_type size() const { return _sys->n_dofs(); }
-
+  ParameterMesh(const FEType & param_type,
+                const std::string & exodus_mesh,
+                const std::vector<std::string> & var_names = {});
+
+  dof_id_type size() const { return _param_dofs; }
+/**
+ * getIndexAndWeight is only used by ParameterMeshFunction
+*/
   void getIndexAndWeight(const Point & pt,
                          std::vector<dof_id_type> & dof_indices,
                          std::vector<Real> & weights) const;
 
   void getIndexAndWeight(const Point & pt,
                          std::vector<dof_id_type> & dof_indices,
                          std::vector<RealGradient> & weights) const;
+/**
+ * getParameterValues is only used by ParameterMeshOptimization
+*/
+  std::vector<Real> getParameterValues(std::string var_name, unsigned int timestep) const;
 
 protected:
```

Comment:
whole class could use a docstring upgrade

why does it need an equation system, its own communicator and output

Reply:
I updated the doxygen.  The equation system ensures the ordering of the parameters are the same between the main and sub-apps.  It doesn't output anything, it is only reading from an exodus mesh that is already created.  The communicator is to make sure this is only done on every processor because the optimization executioner is serial.